odl/discr/lp_discr.py

```diff
@@ -228,6 +234,12 @@ def exponent(self):
         """The exponent ``p`` in ``L^p``."""
         return self._exponent
 
+    @property
+    def vector_field_space(self):
+        """vector field space."""
+        self.real_space = self.astype(self._real_dtype)
+        return ProductSpace(self.real_space, self.ndim)
```

Comment:
I think we should generalize this (and simplify the code). There is nothing special about vector fields, which are just rank-1 tensors. We should allow arbitrary rank. In practice, this means that we make this a method with a `rank=1` optional parameter and raise `NotImplementedError` for anything else than 0 or 1 right now. That's not great, but at least the change to general tensors won't break API then.